src/main/java/com/google/neighborgood/servlets/TaskServlet.java

```diff
@@ -160,25 +184,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String userId = userService.getCurrentUser().getUserId();
 
-    // Create an Entity that stores the input comment
-    Entity taskEntity = new Entity("Task");
-    taskEntity.setProperty("userId", userId);
-    taskEntity.setProperty("detail", taskDetail);
-    taskEntity.setProperty("timestamp", creationTime);
-    taskEntity.setProperty("reward", rewardPts);
-    taskEntity.setProperty("status", "OPEN");
-    taskEntity.setProperty("Owner", userId);
-    taskEntity.setProperty("Helper", "N/A");
-    taskEntity.setProperty("Address", "4xxx Cxxxxx Avenue, Pittsburgh, PA 15xxx");
-    taskEntity.setProperty("zipcode", "98033");
-    taskEntity.setProperty("country", "United States");
-    taskEntity.setProperty("category", "misc");
-
+    // Creates current user entity key to include as the task's parent
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(taskEntity);
+    Key userEntityKey = KeyFactory.createKey("UserInfo", userId);
 
-    // Redirect back to the user page.
-    response.sendRedirect("/user_profile.jsp");
+    try {
```

Comment:
The below lines of code are mostly the same, I just moved everything into a try-catch statement due to the addition of line 192.